cosmo_tester/test_suites/stress_test_openstack/many_deployments_to_single_manager_parallel_test.py

```diff
@@ -0,0 +1,250 @@
+                     "manager_memory_available":
+                         manager_memory_available,
+                     "number_of_active_nodes":
+                         number_of_active_nodes,
+                     "blueprints": self.blueprints,
+                     "deployments": self.deployments,
+                     "installed": self.installed,
+                     "manager_cpu_usage":
+                         self.get_manager_cpu_usage()
+                     }
+            self.state_dict.update({
+                time.time(): state})
+            self.logger.info("current_state: {0}".format(state))
+            time.sleep(5)
+            self.logger.info(
+                "number_of_active_threads {0}".format(self.active_threads))
+
+    def run_thread_with_this(self, func, args):
+        self.active_threads = self.active_threads + 1
+        t = Thread(target=func,
+                   args=args or [])
+        t.start()
+        return t
+
+    def wait_until_all_deployment_executions_end(self, deployment_id):
+        start_time = time.time()
+        while len([execution for execution in self.client.executions.list(
+                deployment_id=deployment_id)
+                if execution["status"] not in Execution.END_STATES]) > 0:
+            self.logger.info("waiting for executions to end on deployment {0}"
+                             .format(deployment_id))
+            time.sleep(1)
+            if time.time() - start_time > self.execute_timeout:
+                return
+        return
+
+    @retry(wait_exponential_multiplier=1000, wait_exponential_max=20000,
+           stop_max_attempt_number=5)
+    def create_deployment_retry(
+            self, params_dict):
+        deployment_id = params_dict.get('deployment_id')
+        blueprint_id = params_dict.get('blueprint_id')
+        inputs = params_dict.get('inputs', '')
+        result = self.create_deployment(blueprint_id,
+                                        deployment_id,
+                                        inputs)
+        self.active_threads = self.active_threads - 1
+        return result
+
+    @retry(wait_exponential_multiplier=1000, wait_exponential_max=20000,
+           stop_max_attempt_number=100)
+    def upload_blueprint_retry(
+            self,
+            blueprint_id):
+        result = self.upload_blueprint(blueprint_id)
+        self.active_threads = self.active_threads - 1
+        return result
+
+    @retry(wait_exponential_multiplier=1000, wait_exponential_max=20000,
+           stop_max_attempt_number=5)
+    def execute_install_retry(self,
+                              deployment_id=None,
+                              fetch_state=True):
+        self.wait_until_all_deployment_executions_end(
+            deployment_id=deployment_id)
+        result = self.execute_install(deployment_id, fetch_state)
+        self.active_threads = self.active_threads - 1
+        return result
+
+    def many_deployments_stress_test(self):
```

Comment:
we haven't really discussed that. So far it pretty much passes on anything that doesn't crash it. I'm thinking maybe adding a minimum number of deployments and a timeout of 30 minutes. so the success condition is that the test ends on a 30 minutes timeout from test start, and has a minimum of say 50 deployments. sounds ok ?